language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -56,8 +56,11 @@ main :: IO ()
 main = do
     opts@Options{..} <- execParser optionsParserInfo
     dar <- B.readFile optInputDar
-    dalfs <- either fail pure $ DAR.readDalfs $ Zip.toArchive $ BSL.fromStrict dar
-    forM_ ((DAR.mainDalf dalfs, optMainPackageName) : map (, Nothing) (DAR.dalfs dalfs)) $ \(dalf, mbPkgName) -> do
+    let archive = Zip.toArchive $ BSL.fromStrict dar
+    dalfs <- either fail pure $ DAR.readDalfs archive
+    DAR.DalfManifest{packageName, ..} <- either fail pure $ DAR.readDalfManifest archive
+    packageName <- pure $ maybe packageName (const optMainPackageName) optMainPackageName
```

Comment:
```suggestion
    packageName <- pure $ fromMaybe packageName optMainPackageName
```

Reply:
I don't think that's right. `packageName` is itself a `Maybe`.

Reply:
Oh good point, you want `optMainPackageName <|> packageName`

Reply:
Ooh! Awesome phrasing. Will adjust!